balancer-js/src/modules/pools/proportional-amounts/index.ts

```diff
@@ -0,0 +1,54 @@
+import { parseUnits } from '@ethersproject/units';
+import { BigNumber } from '@ethersproject/bignumber';
+
+/**
+ * Calculates the proportional amounts of tokens in relation to a given token and amount.
+ * Useful for calculating the amounts of tokens to be sent to a pool when joining or swapping.
+ * When using proportional amounts to join a pool the price impact will be minimal.
+ *
+ * @param pool - Pool object
+ * @param token - Token address in relation to which the amounts are calculated
+ * @param amount - Amount of token
+ * @returns Object with tokens and amounts
+ * @example
+ * const pool = {
+ *   id: '0x0000',
+ *   tokens: [
+ *     { address: '0x1234', balance: '10' },
+ *     { address: '0x5678', balance: '20' }
+ *   ]
+ * }
+ *
+ * const { tokens, amounts } = proportionalAmounts(pool, '0x1234', '1000000000000000000')
+ */
+export const proportionalAmounts = (
+  pool: {
+    id: string;
+    tokens: { address: string; balance: string; decimals?: number }[];
+  },
+  token: string,
+  amount: string
+): {
+  tokens: string[];
+  amounts: string[];
+} => {
+  const noBpt = pool.tokens.filter(
+    (t) => !pool.id.toLowerCase().includes(t.address.toLowerCase())
+  );
+  const index = noBpt.findIndex(
```

Comment:
I feel that this code is a bit hard to read. I'd change the variable names to make it more readable:
`noBpt` -> `tokensWithoutBpt`
`index` -> `referenceTokenIndex`
`amounts` -> `proportionalAmounts`

Reply:
great names, feel free to suggest

Reply:
done